src/store/actions.ts

```diff
@@ -0,0 +1,133 @@
+import { ActionTree } from 'vuex';
+import Web3EthContract from "web3-eth-contract";
+import Web3 from "web3";
+import { StateInterface } from './index';
+import { Config } from '../types/config';
+import { ConnectPayload } from './mutations';
+
+const UNRECOGNIZED_CHAIN_ID_ERROR_CODE = 4902;
+
+/**
+ * Switches to network given by chainConfig or creates a new network configuration if needed.
+ * @param ethereum Web3 provider.
+ * @param chainConfig Network configuration.
+ */
+const switchNetwork = async (ethereum: any, chainConfig: Config) => {
+  const networkId: number = await ethereum.request({
+    method: "net_version",
+  });
+
+  if (networkId != chainConfig.network.id) {
+    const chainIdHex = `0x${chainConfig.network.id.toString(16)}`;
+
+    try {
+      await ethereum.request({
+        method: "wallet_switchEthereumChain",
+        params: [{ chainId: chainIdHex }]
+      });
+    } catch (error) {
+      if (error.code == UNRECOGNIZED_CHAIN_ID_ERROR_CODE ) {
```

Comment:
```
(local var) error: unknown
Object is of type 'unknown'.ts(2571)
```

Type error

Reply:
Weird. I can't reproduce. Let me check.

Reply:
It must have something to do with your type script version. I think I fixed the error. Please check again